src/nota/ui.cljs

```diff
@@ -19,26 +20,39 @@
 
 (def ui-top-router (comp/factory TopRouter))
 
+(defn toggleTheme [theme]
+  (set! (.. js/document -documentElement -className) theme))
+
 (defsc Header [_this {:keys [list-pages]}]
-  {}
-  (dom/header
-   (dom/nav
-    (map ui.pages/ui-list-page list-pages)
-    (dom/button {:onClick #(routing/route-to! (dr/path-to ui.posts.pagination/PaginatedPosts "list"))}
-                "Blog")
-    (dom/button :.float-right
-                {:onClick #(js/window.open "https://github.com/rafaeldelboni/nota" "_blank")}
-                "Source"))))
+  {:use-hooks? true}
+  (let [[theme change-theme] (hooks/use-state "dark")]
+    (js/console.log theme)
+    (dom/header
+     (dom/nav {:class "nota-nav"}
+      (map ui.pages/ui-list-page list-pages)
+      (dom/button {:onClick #(routing/route-to! (dr/path-to ui.posts.pagination/PaginatedPosts "list"))
+                   :class "nota-btn"}
+                  "Blog")
+      (dom/button {:onClick #(js/window.open "https://github.com/rafaeldelboni/nota" "_blank")
+                   :class "nota-btn nota-btn--source"}
+                  "Source")
+      (dom/button {:class "nota-btn nota-btn--theme"
+                   :onClick #(do
```

Comment:
This doesn't need to be in a mutation?

Reply:
I don't think it would be necessary as we are making the "transaction" using the react hook. Don't know if it makes any difference.